### DISCUSSION TITLE
New front panel

### DISCUSSION THREAD

**Discussion by gcormier** [gcormier/megadesk#93] [General]
For those who have a non-functional front panel, damaged ribbon cable, etc

![image](https://user-images.githubusercontent.com/8388983/159293206-812f4c54-f04c-4502-9f2b-c09b208bf643.png)

The PCB is 0.8mm thick and is flush with the face.

- There's room to add another button, but adding 2 more might be a bit tight
- A 3d-printed face could be an option that covers the ribbon/ffc connector. It could snap into the key-holes on both sides
- The key is no longer functional/required

https://github.com/gcormier/megadesk/tree/master/buttons

**Comment by gcormier** [comment]
Big thanks to Gina ( https://github.com/foosel ) for sending me a spare case that I ripped apart to get the dimensions and do all the test fits with 3D-printed prototypes.

**Comment by gcormier** [comment]
Another thought - I could make the board double sided. One has 2 buttons, the other has 3, and one could decide which side to actually use. FFC footprint on both sides obviously.

v8 and onwards have 2 extra pins used by the FFC ribbon that go to this board. So it the max is 4 buttons.

**Comment by gcormier** [comment]
I realize a dual sided board has issues with the slot needing to be cutout twice. But it's pretty ugly.

![image](https://user-images.githubusercontent.com/8388983/160243648-1eaa18d4-1e7d-48d1-acc5-022fbb30f239.png)

This can slide over to cover things, at the expense of a slightly thicker front bezel.

![image](https://user-images.githubusercontent.com/8388983/160243744-ea2c843a-57b4-4511-a656-1ea07a1d8c8d.png)

![image](https://user-images.githubusercontent.com/8388983/160243754-1941ac62-4fe6-4a0b-9bdc-f74bcbdc637f.png)

It uses the existing safety pin as a fixture. A dot or two of CA could be used if something more permanent is desired.

**Comment by gcormier** [comment]
![image](https://user-images.githubusercontent.com/8388983/184501425-4a4658b1-5ba5-4c0a-a89d-fcd888345149.png)

![image](https://user-images.githubusercontent.com/8388983/184501434-b7ec2cfa-3667-4171-a6a9-713c7e8eb458.png)

With the buttons.

A small bend is necessary in the ribbon cable since the opening in the grey housing does not line up with the FFC connector on megadesk.

![image](https://user-images.githubusercontent.com/8388983/184501524-50c091d8-2ee5-436f-be87-770972f46bbd.png)

![image](https://user-images.githubusercontent.com/8388983/184501541-66f67c0c-f695-4d93-9cb3-5c5bebabd9b7.png)

FFC is a 2" (51mm) length, such as https://www.digikey.ca/en/products/detail/assmann-wsw-components/AFFC-100-10-051-11/6570435
3" could be used as well, makes for a bigger loop but there is room to fold it in half.

Putting the FFC into the button-PCB is a bit delicate. I have inserted it first, then fished it through the slot. I think if I order any more, I would route out the slot completely since it's now hidden by this case. Or perhaps I could leave it as-is, and then if it's going to be hidden, then one can just snip out the piece on the purple lines:

![image](https://user-images.githubusercontent.com/8388983/184501669-ebe8f41d-0a16-495a-8cca-3ce813497d2a.png)

**Comment by ryanmatthews** [comment]
I found this because I thought my buttons had died completely. I ultimately resolved the issue with the solution in this post [https://github.com/gcormier/megadesk/discussions/114](url), but I detest the feel of the original buttons, and would absolutely buy this kit if you offered it. Space permitting I would say the buttons could even be bigger. I find the issue with the original is that it's not absolutely clear if the button is pressed or not pressed - having a more assured/positive feeling would be great.

**Reply by gcormier** [reply]
Do you have access to a 3d printer?

**Comment by ryanmatthews** [comment]
I don’t own one, but I do have a friend who has a couple of different ones.

**Comment by gcormier** [comment]
The files to 3D-print if you want the housing are here 

https://github.com/gcormier/megadesk/tree/master/buttons

**Comment by ryanmatthews** [comment]
Hey Greg,

I know Canada Post is slow at the holidays, but I just wanted to check in and see if you had a chance to solder/mail this. I’m incredibly excited to try the mod.

Ryan
